modules/node-shared/src/main/scala/org/tessellation/node/shared/http/routes/MetagraphMessageRoutes.scala

```diff
@@ -0,0 +1,38 @@
+package org.tessellation.node.shared.http.routes
+
+import cats.effect.Async
+import cats.syntax.flatMap._
+import cats.syntax.option._
+
+import org.tessellation.node.shared.infrastructure.snapshot.services.MetagraphMessageService
+import org.tessellation.node.shared.infrastructure.snapshot.storage.MetagraphMessageStorage
+import org.tessellation.routes.internal._
+import org.tessellation.security.{Hasher, SecurityProvider}
+
+import eu.timepit.refined.auto._
+import org.http4s.HttpRoutes
+import org.http4s.dsl.Http4sDsl
+
+class MetagraphMessageRoutes[F[_]: Async: SecurityProvider: Hasher](
+  service: MetagraphMessageService[F],
+  storage: MetagraphMessageStorage[F]
+) extends Http4sDsl[F]
+    with PublicRoutes[F] {
+
+  protected val prefixPath: InternalUrlPrefix = "/currency"
+
+  protected val public: HttpRoutes[F] = HttpRoutes.of[F] {
+    case POST -> Root / "signing" / "owner" / dagAddress =>
```

Comment:
Shouldn't it receive already signed message? 🤔

Reply:
Implementation changed to process payload